common/src/primitives/amount.rs

```diff
@@ -317,6 +323,27 @@ mod tests {
         assert_eq!(Amount { val: IntType::MAX } + Amount { val: 1 }, None);
     }
 
+    #[test]
+    fn sum_some() {
+        let amounts = vec![Amount { val: 1 }, Amount { val: 2 }, Amount { val: 3 }];
+        assert_eq!(
+            amounts.into_iter().sum::<Option<Amount>>(),
+            Some(Amount { val: 6 })
+        );
+    }
+
+    #[test]
+    fn sum_overflow() {
```

Comment:
Perhaps also add a test for an empty vector

Reply:
Done.